internal/controllers/gateway/tcproute_controller.go

```diff
@@ -361,14 +361,12 @@ func (r *TCPRouteReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	// we can update the object status to indicate that it's now properly linked
 	// to the configured Gateways.
 	debug(log, tcproute, "Ensuring status contains Gateway associations")
-	statusUpdated, err := r.ensureGatewayReferenceStatusAdded(ctx, tcproute, gateways...)
+	shouldRequeue, err := r.ensureGatewayReferenceStatusAdded(ctx, tcproute, gateways...)
 	if err != nil {
 		// don't proceed until the statuses can be updated appropriately
 		return ctrl.Result{}, err
 	}
-	if statusUpdated {
-		// if the status was updated it will trigger a follow-up reconciliation
-		// so we don't need to do anything further here.
+	if shouldRequeue {
```

Comment:
I think, it may be non-obvious for someone why it will requeue when empty `ctrl.Result{}` is returned